src/scripts/components/LanguagePanel/Content.jsx

```diff
@@ -1,33 +1,75 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 
 import CreateTranslations from './CreateTranslations';
 import useEditPost from '../../hooks/useEditPost';
-import { getQueryVar } from '../../services/searchQuery';
 import getUBBSetting from '../../services/settings';
 import LangContext from '../../contexts/LangContext';
 import PostLanguage from './PostLanguage';
 import ListTranslations from './ListTranslations';
 import UnlinkTranslations from './UnlinkTranslations';
 import LinkTranslations from './LinkTranslations';
+import { linkPost } from '../../services/requests';
+import { getQueryVar } from '../../services/searchQuery';
+
+// Because Gutenberg.
+let sourceToLink       = new URLSearchParams(window.location.search).get("ubb_source");
+let unsubscribeLinking = null;
 
 const Language = () => {
-	const [postId, setPostId] = useState(getQueryVar('post'));
-	const { data, refetch, isLoading, isError } = useEditPost(postId);
-	const [, setIsSavingMetaboxes] = useState(
-		wp?.data?.select('core/edit-post')?.isSavingMetaBoxes() || false
-	);
+	const postId = wp?.data?.select('core/editor')?.getCurrentPostId() ?? getQueryVar('post');
+	const { data, refetch, isLoading, isError, setIsLoading, setIsError } = useEditPost(postId);
 
-	wp.data.subscribe(() => {
-		setIsSavingMetaboxes((prev) => {
-			const current = wp.data
-				.select('core/edit-post')
-				.isSavingMetaBoxes();
-			if (prev && !current) {
-				setPostId(wp.data.select('core/editor').getCurrentPostId());
+	// Link post to source post if ubb_source is present in the URL.
+	useEffect(() => {
+
+		// Don't run on classic editor.
+		if ( postId === undefined ) {
+			return;
+		}
+
+		// Don't subscribe if sourceToLink is not set.
+		if ( ! sourceToLink ) {
+			return;
+		}
+
+		// Save the unsubscribe function to be able to unsubscribe later.
+		unsubscribeLinking = wp.data.subscribe(() => {
+			// Don't run if sourceToLink is not set.
+			if ( ! sourceToLink ) {
+				return;
 			}
-			return current;
+
+			// Check if post save request was successful and if the post is new.
+			const didPostSaveRequestSucceed = wp.data.select('core/editor').didPostSaveRequestSucceed();
+			const isEditedPostNew           = wp.data.select('core/editor').isEditedPostNew();
+			if ( ! didPostSaveRequestSucceed || isEditedPostNew ) {
+				return;
+			}
+
+			// Set loading to keep the user from interacting with the interface.
+			setIsLoading(true);
+
+			// Try to link the post to the source post.
+			linkPost(wp.data.select('core/editor').getCurrentPostId(), sourceToLink)
+				.then(() => {
+
+					// Refetch data to update the interface.
+					refetch();
+				})
+				.catch(() => {
+
+					// Set error to show the user that something went wrong.
+					setIsLoading(false);
```

Comment:
This should come in a `.then()` after `.catch()` to always be called regardless of the outcome of the request.